### DISCUSSION TITLE
Alternate Internal Link Rendering

### DISCUSSION THREAD

**Discussion by LeifAndersen** [Zettlr/Zettlr#4029] [General]
Hey, I have all of my notes in files with the name `<long-id>.md`, and refer to them as `[[<long-id]]` in the text of other notes. However, each of these notes has a human readable title (and/or first header element), which is clearly seen when viewing the files from the left menu.

So for example, I may have a file called:

Filename: 0000-0000-0000-0001.md
Contents:
```
# Horses

Some rad notes about horses...
```

And then in another note, I may say:

```
Ponies are like small [[0000-0000-0000-0001]].
```

By doing this, I'm able to change the title of `Horses` at a later date, without breaking any of the old references I previously had there.

When I mouse over `[[0000-0000-0000-0001]]`, I see a popup that says `Horses`, and the files list I see a file that says `Horses`. What I'd like is to have the markdown render as `Horses` as well. I still want the full uuid to be stored in the file, just to be able to view it nicely, somewhat like how it renders nice looking citations, while still keeping the full citation ID in the text.

Is there any existing way to do this? And if not, would anyone be amenable to me making a PR to add that functionality?

**Comment by nathanlesage** [comment]
There's currently no way to do this, but something is planned. I'm waiting and coordinating here with the folks over at Pandoc as they plan to integrate something like that as well.

**Reply by LeifAndersen** [reply]
Does zettr use pandoc for the internal markdown editor? (I had previously thought that was just for exporting.)

**Reply by nathanlesage** [reply]
It is just for exporting, but if the editor shows one thing, but then export something else, that will cause confusion. Therefore, I try to align Zettlr's internal representation as close as possible with Pandoc's export.

**Comment by LeifAndersen** [comment]
Hmm...alternatively, looking into this a bit more, `renameFile` as provided by `source/app/service-providers/commands/file-rename.ts` would alternatively work for me. Is there any way I can get zettlr to use that renameFile rather than the one in `source/app/service-providers/fsal/index.ts`?
